### PR TITLE
Make logger generic for `XcUITestDriver`

### PR DESCRIPTION
**Pull request by amanjeetsingh150**
- Renames IOSUiTestRunner to `XCUITestDriver` as an interface
- Moves away from using `DebugLogStore` directly in driver. Makes logger interface instead

### CODE REVIEW

File: maestro-client/src/main/java/maestro/debuglog/IOSDriverLogger.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package maestro.debuglog
+
+import ios.logger.Logger
+import maestro.drivers.IOSDriver
+
+class IOSDriverLogger : Logger {
+
+    private val logger by lazy { DebugLogStore.loggerFor(IOSDriver::class.java) }
```
Comment (axelniklasson) [inline maestro-client/src/main/java/maestro/debuglog/IOSDriverLogger.kt:8]: Should this be `IOSDriverLogger`?
```suggestion
    private val logger by lazy { DebugLogStore.loggerFor(IOSDriverLogger::class.java) }
```

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt [hunk 2]
```diff
@@ -70,18 +73,13 @@ class IOSDriver(
 
     private fun ensureXCUITestChannel() {
         logger.info("[Start] Uninstalling xctest ui runner app on ${iosDevice.deviceId}")
-        IOSUiTestRunner.uninstall()
+        xcUiTestDriver.uninstall()
```
Comment (axelniklasson) [inline maestro-client/src/main/java/maestro/drivers/IOSDriver.kt:76]: Do we need any logs in this method like we had before?

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt [hunk 3]
```diff
@@ -212,7 +209,7 @@ class IOSDriver(
 
         val contentDescriptorResult = iosDevice.contentDescriptor(
             resolvedAppId ?: throw IllegalStateException("Failed to get view hierarchy, app id was not resolvedGetRunningAppRequest.kt")
-        )
+        ) { errorMessage -> logger.info(errorMessage) }
```
Comment (axelniklasson) [inline maestro-client/src/main/java/maestro/drivers/IOSDriver.kt:212]: Should we instead add an `error` function and call `logger.error` here?

**Review by axelniklasson — APPROVED** [review]

File: maestro-client/src/main/java/maestro/debuglog/IOSDriverLogger.kt (hunk 1 above)
Comment (amanjeetsingh150) [reply maestro-client/src/main/java/maestro/debuglog/IOSDriverLogger.kt]: Nice catch! Done ✅

Comment (amanjeetsingh150) [reply maestro-client/src/main/java/maestro/debuglog/IOSDriverLogger.kt]: Nice catch! Done

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt (hunk 2 above)
Comment (amanjeetsingh150) [reply maestro-client/src/main/java/maestro/drivers/IOSDriver.kt:76]: The logs are still there they are just shifted to the `XcUiTestDriver` class

File: maestro-client/src/main/java/maestro/drivers/IOSDriver.kt (hunk 3 above)
Comment (amanjeetsingh150) [reply maestro-client/src/main/java/maestro/drivers/IOSDriver.kt:212]: Unfortunately, there is no logger error for the `DebugLogStore` class. I'll start a thread on this and let's solve that separately.

**Pull request merged**